dali/test/python/test_dali_tf_dataset_eager.py

```diff
@@ -325,6 +331,20 @@ def test_tf_experimental_inputs_disabled():
                         input_datasets={"test" : tf.data.Dataset.from_tensors(np.int32([42, 42]))})
 
 
+# Test if the ValueError is raised for external source with source.
+@raises(ValueError)
+def test_tf_experimental_source_disabled():
+    pipe = Pipeline(10, 4, 0)
```

Comment:
Maybe you can manually test if exception is raised and if the message is correct?

Reply:
:+1:  assert_raises is a much finer tool

Reply:
I won't do it unless we decide on changing this in the whole code base and decide on some more convenient way.
Swapping one decorator, to with scopes with additional checks doesn't sound like nice way of doing things.

It also couples tests with the implementation if we go overboard with those checks.

Reply:
Well, with anything other than RuntimeError it's passable. With RuntimeError not really. Just 2 days ago I found several errors by inspecting messages (otherwise it the pipeline did total mess and still raised RuntimeError).